setup.py

```diff
@@ -27,7 +27,12 @@ def read_req_file(req_type):
     long_description_content_type="text/markdown",
     install_requires=read_req_file("install"),
     extras_require={
-        "dev": read_req_file("dev"),
+        "dev": read_req_file("dev")
+        + [
```

Comment:
this is a compromise of the solution, the `pip-compile` has no flexibility of locking down a specific version based on python version.